lastgersync/processor.go

```diff
@@ -95,6 +112,44 @@ func (p *processor) ProcessBlock(ctx context.Context, block sync.Block) error {
 	return nil
 }
 
+// handleGERInsertion inserts the given global exit root entry to `imported_global_exit_root`
+func (*processor) handleGERInsertion(tx db.Txer, event *Event, blockNum uint64) error {
+	gerInfoWithBlockNum := &gerInfoWithBlockNum{
+		GlobalExitRoot:  event.GERInfo.GlobalExitRoot,
+		L1InfoTreeIndex: event.GERInfo.L1InfoTreeIndex,
+		BlockNum:        blockNum,
+	}
+
+	if err := meddler.Insert(tx, "imported_global_exit_root", gerInfoWithBlockNum); err != nil {
+		return fmt.Errorf("failed to insert GER entry (value=%x, block=%d): %w",
+			event.GERInfo.GlobalExitRoot, blockNum, err)
+	}
+	return nil
+}
+
+// handleGEREvent either inserts or removes the global exit root entry from `imported_global_exit_root` table,
+func (p *processor) handleGEREvent(tx db.Txer, event *GEREvent) error {
+	if event.IsRemove {
+		_, err := tx.Exec(deleteGERSql, event.GlobalExitRoot.Hex())
+		if err != nil {
+			return fmt.Errorf("failed to remove global exit root %s: %w", event.GlobalExitRoot.Hex(), err)
+		}
+	} else {
+		err := meddler.Insert(tx,
```

Comment:
You could use this function here: `handleGERInsertion`. You would just need to change it to receive `GEREvent`.